src/components/Header/Header.tsx

```diff
@@ -0,0 +1,93 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import {
+  FC, FormEvent, useCallback, useState,
+} from 'react';
+import { Todo } from '../../types/Todo';
+import { Errors } from '../../utils/enums';
+import { TodoData } from '../../types/TodoData';
+import { createTodo } from '../../api/todos';
+
+const USER_ID = 10284;
+
+interface Props {
+  todos: Todo[];
+  onAdd: (newTodo: Todo) => void;
+  setError: (error: null | Errors) => void;
+  setTempTodo: (newTodo: Todo | null) => void;
+  setIsLoading: (boolean: boolean) => void
+  loadTodos: () => void;
+}
+
+export const Header:FC<Props> = ({
+  todos,
+  onAdd,
+  setError,
+  setTempTodo,
+  setIsLoading,
+  loadTodos,
+}) => {
+  const [query, setQuery] = useState('');
+  const [isDisabled, setIsDisabled] = useState(false);
+
+  const handleSubmit = useCallback(async (event: FormEvent) => {
+    event.preventDefault();
```

Comment:
Add the missing dependencies 'setError', 'setTempTodo', 'setIsLoading', 'loadTodos', and 'onAdd' to the dependency array of useCallback.